test/grpc/error_test.exs

```diff
@@ -0,0 +1,33 @@
+defmodule GRPC.RPCErrorTest do
+	use ExUnit.Case
+
+  alias GRPC.RPCError
+
+  describe "is_rpc_error/2 guard clause" do
+    import GRPC.RPCError
+
+    test "returns true for GRPC.RPCError structs with same status" do
+      exception = GRPC.RPCError.new(:unknown)
+
+      assert is_rpc_error(exception, 2)
+    end
+
+    test "returns false for GRPC.RPCError structs with different status" do
+      exception = GRPC.RPCError.new(:unknown)
+
+      refute is_rpc_error(exception, 1)
+    end
+
+    test "returns false for others types of input" do
+      exception = %{status: :unknown}
+
+      refute is_rpc_error(exception, 2)
```

Comment:
```suggestion
      exception = %{status: :unknown}

      refute is_rpc_error(exception, 2)
      refute is_rpc_error(:other_type, 2)
```

Reply:
Just to show that non-maps also fail the guard

Reply:
For using the guard clause with types other than `map()`, we have the following warning

![image](https://github.com/elixir-grpc/grpc/assets/179497/805d6c27-7f92-41ce-b65e-0450b4ba98d0)

I think this is because we are using the access method for the status key in the guard clause 🤔